esx_service/utils/auth_api.py

```diff
@@ -399,6 +456,10 @@ def _tenant_create(name, description="", vm_list=None, privileges=None):
         logging.debug("_tenant_create: vms=%s", vms)
         vms_uuid_list = [(vm_id) for (vm_id, vm_name) in vms]
 
+    error_info = check_default_datastore(default_datastore)
+    if error_info:
+        return error_info
```

Comment:
should return error_info, None.
Plus we need a test case to capture this error.